src/Pages/Main/Main.js

```diff
@@ -1,7 +1,222 @@
-import React from 'react';
+import { useState, useEffect, useCallback } from 'react';
+import { Link } from 'react-router-dom';
+import { useOutsideClick } from '../../utils/inputOutsideClick';
+import SearchModal from '../../Components/SearchModal/SearchModal';
+import RecommedProduct from '../../Components/RecommedProduct/RecommedProduct';
+import styled from 'styled-components/macro';
+import { icons } from '../../icons';
 
-const Main = () => {
-  return <div></div>;
-};
+function Main() {
+  const [inputValue, setInputValue] = useState('');
+  const [filterData, setFilterData] = useState([]);
+  const [randomData, setRandomData] = useState([]);
+  const { isActive, setIsActive, inputEl } = useOutsideClick();
+
+  const changeInputValue = useCallback(e => {
+    setInputValue(e.target.value);
+  }, []);
+
+  useEffect(() => {
+    fetch(`http://localhost:3000/data/searchFilterData.json`)
+      .then(res => res.json())
+      .then(res => {
+        let arr = [];
+        for (let i = 0; i < res.result.length; i++) {
+          if (res.result[i].title.includes(inputValue) && inputValue) {
+            arr.push(res.result[i]);
+          } else if (!inputValue) {
+            arr.slice(0);
+          }
+        }
+        console.log(arr);
+        setFilterData(arr);
+      });
```

Comment:
```suggestion
      .then(res => {
        const filteredInputs = res.result.filter(({ title }) => {
          return inputValue && title.includes(inputValue)
        })
        
        setFilterData(filteredInputs);
      });
```
- `arr.slice(0)`가 하는 역할은 무엇인가요? slice()는 배열에서 특정 인덱스를 잘라낸 복사본을 리턴하는 함수인데 리턴 값이 쓰이지 않고 있고, 삭제하셔도 좋을 것 같습니다.
- 배열을 순회하며 특정 기준에 맞는 요소들만 골라낸 뒤 새로운 배열로 리턴하는 동작이라면 filter가 보다 적합할 것 같습니다.

Reply:
삭제했을 때 filterData에는 값이 남아 있어서 화면상에 보이게 되지만 값이 없을 때는 보이고 싶지않아서 넣어뒀는데 그래도 안되나요?